packages/core/src/rules/common/no-required-schema-properties-undefined.ts

```diff
@@ -0,0 +1,41 @@
+import { Oas2Rule, Oas3Rule } from '../../visitors';
+import { Oas3Schema, Oas3_1Schema } from '../../typings/openapi';
+import { Oas2Schema } from 'core/src/typings/swagger';
+import { UserContext } from 'core/src/walk';
+import { isRef } from '../../ref-utils';
+
+export const NoRequiredSchemaPropertiesUndefined: Oas3Rule | Oas2Rule = () => {
+  return {
+    Schema: {
+      enter(
+        schema: Oas3Schema | Oas3_1Schema | Oas2Schema,
+        { location, report, resolve }: UserContext
+      ) {
+        const allProperties = schema.properties ?? {};
+
+        // Skip oneOf/anyOf as it's complicated to validate it right now.
+        // We need core support for checking contrstrains through those keywords.
+        // Right now we only resolve basic allOf keyword nesting.
+        if (schema.allOf) {
+          schema.allOf.forEach((nestedSchema: Oas3Schema | Oas3_1Schema | Oas2Schema) => {
+            if (isRef(nestedSchema)) {
+              const resolved = resolve(nestedSchema).node as Oas3Schema | Oas3_1Schema | Oas2Schema;
```

Comment:
resolved can be another allOf we need to do it recursivelly

Reply:
but also we should make sure we won't get stack overflow if nested allOf's are recursive (which can happen in theory

Reply:
Yeah, I suggested we stop at the first level for the sake of simplicity. Do you think it makes much sense to fix this on the rule level? It looks like other rules also need something similar.

Reply:
Code was refactored to cover nested `allOf` schemas recursively, including stack overflow guard